cardano-node/src/Cardano/Node/Tracing/Tracers/Diffusion.hs

```diff
@@ -53,6 +61,274 @@ instance MetaTrace tr => MetaTrace (WithMuxBearer peer tr) where
+      , "msg"  .= String "Timed out reading SDU"
+      ]
+    forMachine _dtal MuxTraceSDUWriteTimeoutException = mconcat
+      [ "kind" .= String "MuxTraceSDUWriteTimeoutException"
+      , "msg"  .= String "Timed out writing SDU"
+      ]
+    forMachine _dtal (MuxTraceStartEagerly mid dir) = mconcat
+      [ "kind" .= String "MuxTraceStartEagerly"
+      , "msg"  .= String "Eagerly started"
+      , "miniProtocolNum" .= String (showT mid)
+      , "miniProtocolDir" .= String (showT dir)
+      ]
+    forMachine _dtal (MuxTraceStartOnDemand mid dir) = mconcat
+      [ "kind" .= String "MuxTraceStartOnDemand"
+      , "msg"  .= String "Preparing to start"
+      , "miniProtocolNum" .= String (showT mid)
+      , "miniProtocolDir" .= String (showT dir)
+      ]
+    forMachine _dtal (MuxTraceStartedOnDemand mid dir) = mconcat
+      [ "kind" .= String "MuxTraceStartedOnDemand"
+      , "msg"  .= String "Started on demand"
+      , "miniProtocolNum" .= String (showT mid)
+      , "miniProtocolDir" .= String (showT dir)
+      ]
+    forMachine _dtal (MuxTraceTerminating mid dir) = mconcat
+      [ "kind" .= String "MuxTraceTerminating"
+      , "msg"  .= String "Terminating"
+      , "miniProtocolNum" .= String (showT mid)
+      , "miniProtocolDir" .= String (showT dir)
+      ]
+    forMachine _dtal MuxTraceStopping = mconcat
+      [ "kind" .= String "MuxTraceStopping"
+      , "msg"  .= String "Mux stopping"
+      ]
+    forMachine _dtal MuxTraceStopped = mconcat
+      [ "kind" .= String "MuxTraceStopped"
+      , "msg"  .= String "Mux stoppped"
+      ]
+#ifdef os_HOST_linux
+    forMachine _dtal (MuxTraceTCPInfo StructTCPInfo
+            { tcpi_snd_mss, tcpi_rcv_mss, tcpi_lost, tcpi_retrans
+            , tcpi_rtt, tcpi_rttvar, tcpi_snd_cwnd }
+            len) =
+      [ "kind" .= String "MuxTraceTCPInfo"
+      , "msg"  .= String "TCPInfo"
+      , "rtt"  .= String (show (fromIntegral tcpi_rtt :: Word))
+      , "rttvar" .= String (show (fromIntegral tcpi_rttvar :: Word))
+      , "snd_cwnd" .= String (show (fromIntegral tcpi_snd_cwnd :: Word))
+      , "snd_mss" .= String (show (fromIntegral tcpi_snd_mss :: Word))
+      , "rcv_mss" .= String (show (fromIntegral tcpi_rcv_mss :: Word))
+      , "lost" .= String (show (fromIntegral tcpi_lost :: Word))
+      , "retrans" .= String (show (fromIntegral tcpi_retrans :: Word))
+      , "length" .= String (showT len)
+      ]
+#else
+    forMachine _dtal (MuxTraceTCPInfo _ len) = mconcat
+      [ "kind" .= String "MuxTraceTCPInfo"
+      , "msg"  .= String "TCPInfo"
+      , "len"  .= String (showT len)
+      ]
+#endif
+
+    forHuman MuxTraceRecvHeaderStart =
+      "Bearer Receive Header Start"
+    forHuman (MuxTraceRecvHeaderEnd MuxSDUHeader { mhTimestamp, mhNum, mhDir, mhLength }) =
+      sformat ("Bearer Receive Header End: ts:" % prefixHex % "(" % shown % ") " % shown % " len " % int)
+        (unRemoteClockModel mhTimestamp) mhNum mhDir mhLength
```

Comment:
`mhNum` has type `newtype MiniProtocolNum = MiniProtocolNum Word16` if you unpack it you could use `int` formatter or (`"MiniProtocol " % int)`, it might be even faster.